sgt-back-end/index.js

```diff
@@ -0,0 +1,150 @@
+const express = require('express');
+const app = express();
+const pg = require('pg');
+
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/studentGradeTable',
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+app.get('/api/grades', (req, res) => {
+  const sql = `
+  select *
+    from "grades"`;
+    db.query(sql)
+    .then((result => {
+      res.status(200).json(result.rows);
+    }))
+    .catch(err => {
+      res.status(500).json({
+        error: 'Something went wrong'
+      })
+    })
+})
+
+app.use(express.json())
+
+app.post('/api/grades', (req, res) => {
+  if(req.body.name === undefined || req.body.course === undefined || req.body.score === undefined || req.body.score < 1 || req.body.score > 100) {
+    res.status(400).json({
+      error: 'Something is wrong with your request parameters'
+    })
+    return;
+  }
+  const text = `
+  INSERT INTO grades (name, course, score)
+    VALUES($1, $2, $3)`
+  const values = [req.body.name, req.body.course, req.body.score]
+  db.query(text, values)
+  .then((result) => {
+    const sql = `
+    select *
+      from "grades"`;
+    db.query(sql)
+    .then((result) => {
+      res.status(201).json(result.rows[result.rows.length - 1])
+    })
```

Comment:
While `SELECT`ing the last row may work if there's only one `INSERT` happening at a time, there's nothing preventing multiple clients from `POST`ing to `/api/grades` at the same time, so this isn't guaranteed to retrieve the same grade that the client `POST`ed. Instead, make use of `RETURNING` to make sure the `INSERT`ed record is retrieved.